kube-deploy/roles/kube-addons/tasks/addons-dashboard.yml

```diff
@@ -14,7 +14,18 @@
 # limitations under the License.
 #
 
+#For ARM support only
+- name: Checking for ARM support
+  shell: uname -a | grep -i "arm"
+  register: arm_enabled
+  ignore_errors: true
+
 - name: deploying the kubernetes dashboard
   shell: kubectl create -f https://rawgit.com/kubernetes/dashboard/master/src/deploy/kubernetes-dashboard.yaml
-  when: kube_addons_enabled and "{{kube_addons.kube_dashboard is defined}}"
+  when: kube_addons_enabled and "{{kube_addons.kube_dashboard is defined}}" and arm_enabled.rc != 0
+  ignore_errors: true
+
+- name: deploying the kubernetes dashboard - ARM support only
+  shell: curl -sSL https://rawgit.com/kubernetes/dashboard/master/src/deploy/kubernetes-dashboard.yaml | sed "s/amd64/arm64/g" | kubectl create -f -
```

Comment:
The replacement of `arm` or `arm64` should be dependent on architecture.  See other comment.